Scripts/AssemblyInjections/Unity.Entities/EntityCommandBufferInternal.cs

```diff
@@ -0,0 +1,20 @@
+using Unity.Entities;
+
+namespace Anvil.Unity.DOTS.Entities
+{
+    /// <summary>
+    /// Helper class to access the internals in an <see cref="EntityCommandBuffer"/>
+    /// </summary>
+    public static class EntityCommandBufferInternal
+    {
+        /// <summary>
+        /// Returns whether the <see cref="EntityCommandBuffer"/> has been played back or not.
+        /// </summary>
+        /// <param name="ecb">The <see cref="EntityCommandBuffer"/> to check.</param>
+        /// <returns>true if it has been played back, false if not</returns>
+        public static unsafe bool DidPlayback(this EntityCommandBuffer ecb)
+        {
+            return ecb.m_Data == null || ecb.m_Data->m_DidPlayback;
```

Comment:
Right now if `m_Data == null` we return true. It that the intended behaviour we want?

That means `default` will return true as will disposed ECBs.

```suggestion
            return ecb.m_Data != null && ecb.m_Data->m_DidPlayback;
```

May also want to use `IsCreated` instead of the data null check 🤷

Reply:
It is but I guess technically this function does too much. 

I've changed it to just be a function to check if the ECB has been played back and callers of this will do their own check for `IsCreated` as well. 

The reason being that `EntityCommandBufferSystem` will dispose of the ECB after playback so we need to check both conditions for the application level code. If the data is still there, then it wasn't run through the System and we can check if it was played back, if the data isn't there then it was run through the system and we can assume it was played back.

Reply:
Right, yea that's tricky. You really need to understand the context around the question you're asking because in a vacuum you have no idea if the instance is `default` or was disposed

Reply:
Agreed. I think for ECBs in particular, with Unity's current usage, if you're asking this question it's to see if you can clean up after yourself and a `default` or disposed instance will prompt the same behaviour.